packages/features/auth/signup/handlers/calcomHandler.ts

```diff
@@ -52,15 +52,33 @@ async function handler(req: RequestWithUsernameStatus, res: NextApiResponse) {
   if (token) {
     foundToken = await findTokenByToken({ token });
     throwIfTokenExpired(foundToken?.expires);
-    await validateUsernameForTeam({ username, email, teamId: foundToken?.teamId ?? null });
+    username = await validateUsernameForTeam({
```

Comment:
Ensure that the username is the one we obtain from email, even if the user provides a different one in the request.